generators/service-watson-conversation/templates/python/instrumentation.py

```diff
@@ -1,10 +1,28 @@
 from ibmcloudenv import IBMCloudEnv
-from watson_developer_cloud import ConversationV1
+from watson_developer_cloud import AssistantV1
 
-conversation = ConversationV1(
-    username=IBMCloudEnv.getString('watson_conversation_username'),
-    password=IBMCloudEnv.getString('watson_conversation_password'),
-    version='2017-05-26')
 
+params = {
+    'version': '2018-02-16',
+    'url': IBMCloudEnv.getString('watson_conversation_url')
+}
+
+if IBMCloudEnv.getString('watson_conversation_apikey'):
+    iam_url = 'https://iam.stage1.bluemix.net/identity/token' if 'gateway-s.' in params['url'] else 'https://iam.bluemix.net/identity/token'
+    iam_apikey = IBMCloudEnv.getString('watson_conversation_apikey')
+    conversation = AssistantV1(
+        url=IBMCloudEnv.getString('watson_conversation_url'),
+        iam_api_key=iam_apikey,
+        version='2018-02-16')
+else:
+    conversation = AssistantV1(
+    version='2018-02-16',
```

Comment:
we still need to provide `username/password` if the instance is not an IAM instance. To make sure generator still works for CF instances.